lib/gui/app/components/svg-icon.js

```diff
@@ -59,31 +83,52 @@ class SVGIcon extends react.Component {
       // eslint-disable-next-line no-underscore-dangle
       : global.__dirname
 
-    // This means the path to the icon should be
-    // relative to *this directory*.
-    // TODO: There might be a way to compute the path
-    // relatively to the `index.html`.
-    const imagePath = path.join(baseDirectory, 'assets', this.props.path)
+    let svgData = ''
+
+    _.find(this.props.contents, (content) => {
+      const attempt = tryParseSVGContents(content)
 
-    let contents = ''
+      if (attempt) {
+        svgData = attempt
+        return true
+      }
+
+      return false
+    })
 
-    if (_.startsWith(this.props.path, '<')) {
-      contents = this.props.path
-    } else {
-      contents = fs.readFileSync(imagePath, {
-        encoding: 'utf8'
+    if (!svgData) {
+      _.find(this.props.paths, (relativePath) => {
+        // This means the path to the icon should be
+        // relative to *this directory*.
+        // TODO: There might be a way to compute the path
+        // relatively to the `index.html`.
+        const imagePath = path.join(baseDirectory, 'assets', relativePath)
```

Comment:
Regarding the TODO here, I think using `path.relative(from,to)` could work here

Reply:
Let's fix that in a separate PR, as this is just a moved block, the TODO isn't new to this PR.

Reply:
Oh, right – yeah, let's do that then

Reply:
Sweet, is this good to merge then?